crates/iceberg-catalog/src/implementations/postgres/tabular/table/common.rs

```diff
@@ -406,8 +406,7 @@ pub(super) async fn insert_snapshot_refs(
     let _ = sqlx::query!(
         r#"
         WITH deleted AS (
-            DELETE FROM table_refs
-            WHERE table_id = $1 AND table_ref_name = ANY($2::TEXT[])
```

Comment:
Shouldn't we better do `WHERE table_id = $1 AND table_ref_name not in ...`?
This would currently always re-write all refs. Refs are updated in almost every commit, we would then need to touch multiple rows. Also the last_updated_timestamp always gets updated now for all references.

Reply:
yes, much better, thanks!